itest/rust/src/object_tests/object_test.rs

```diff
@@ -527,11 +528,10 @@ fn object_engine_upcast() {
 }
 
 fn ref_instance_id(obj: &Object) -> InstanceId {
-    // SAFETY: raw FFI call since we can't access get_instance_id() of a raw Object anymore, and call() needs &mut.
-    use godot::obj::EngineClass as _;
-
-    let obj_ptr = obj.as_object_ptr();
+    // SAFETY: `Object` is a `#[repr(C)]` struct with its first field being the object pointer.
+    let obj_ptr = unsafe { *(obj as *const Object as *const GDExtensionObjectPtr) };
```

Comment:
I assume the reliance on field order here is due to the fact that the fields themselves are private and thus can't be directly accessed?

Reply:
yeah, i was gonna just add a `#[doc(hidden)]` method to `Object` to do this instead tbh since then i dont need to do this.